### PR TITLE
Fix/ios/imap disabled

### LINKED ISSUE #303: [iOS] Catch the error when Gmail IMAP is disabled
### Reason

When the user has IMAP disabled the app needs to give a specific error message to trigger the UI explanation of how it works

### Description

If the Gmail IMAP is disabled, Microblink IMAP returns a specific error message that should be treated differently form the other messages:

image

We need to catch this error in native code and send the information to the UI.

### Success Criteria

- [ ] if IMAP is disabled in Gmail, the plugin is able to detect it and return the correct error message to the UI.

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by JesseMonteiro**
close #303

### CODE REVIEW

File: ios/Plugin/Email/Email.swift [hunk 1]
```diff
@@ -33,16 +33,27 @@ public class Email {
...
         let email = BRIMAPAccount(provider: .gmailIMAP, email: account.user, password: account.password!)
         DispatchQueue.main.async {
             let rootVc = UIApplication.shared.windows.first?.rootViewController
             BREReceiptManager.shared().setupIMAP(for: email, viewController: rootVc!, withCompletion: { result in
                 BREReceiptManager.shared().verifyImapAccount(email, withCompletion: { success, error in
-                    if success {
-                        onSuccess()
-                    } else {
-                        onError(error.debugDescription)
+                    guard let errorCallback: Int = (error as? NSError)?.code else {
```
Comment (ricardobrg) [inline ios/Plugin/Email/Email.swift:42]: the name of this variable is not clear. Looks like it is a callback, but it is just the error returned.

File: ios/Plugin/Email/Email.swift [hunk 2]
```diff
@@ -92,13 +103,18 @@ public class Email {
         BREReceiptManager.shared().dayCutoff = getDayCutOff()
         Task(priority: .high){
             BREReceiptManager.shared().getEReceipts() {scanResults, emailAccount, error in
-                if(scanResults != nil){
-                    scanResults!.forEach{scanResults in
-                        onReceipt(scanResults)
+                guard let errorCallback = error as? BREReceiptIMAPError else {
```
Comment (ricardobrg) [inline ios/Plugin/Email/Email.swift:106]: samething here with the variable name

File: ios/Plugin/Plugin/Rsp/RspError.swift [hunk 3]
```diff
@@ -36,7 +38,7 @@ public class RspError: Rsp {
      
      - Returns: A dictionary representing the error data in a format suitable for a Capacitor plugin call result.
      */
-    override func toPluginCallResultData() -> [String: Any] {
+    override func toPluginCallResultData() -> PluginCallResultData {
```
Comment (ricardobrg) [inline ios/Plugin/Plugin/Rsp/RspError.swift:41]: Is this type casting necessary? If it is, it is better if done in the plugin layer instead, to decouple this code from capacitor

**Review by ricardobrg — CHANGES_REQUESTED** [review]

File: ios/Plugin/Email/Email.swift (hunk 1 above)
Comment (JesseMonteiro) [reply ios/Plugin/Email/Email.swift]: Resolved

File: ios/Plugin/Email/Email.swift (hunk 2 above)
Comment (JesseMonteiro) [reply ios/Plugin/Email/Email.swift]: Resolved

File: ios/Plugin/Plugin/Rsp/RspError.swift (hunk 3 above)
Comment (JesseMonteiro) [reply ios/Plugin/Plugin/Rsp/RspError.swift]: Resolved

**Review by ricardobrg — APPROVED** [review]

**Pull request merged**